app/src/main/java/com/android/wardrobeManager/ui/images/PolkaDots.java

```diff
@@ -0,0 +1,81 @@
+package com.android.wardrobeManager.ui.images;
+
+import android.graphics.Bitmap;
+import java.util.Map;
+
+public class PolkaDots implements DesignFilterManager.DesignFilter {
+
+    protected Map<String, DesignFilterManager.DesignFilter> filters;
+    private int spacing = 80;
+    private int dotRadius = 10;
+    private int colorIndex = 0;
+
+    public Bitmap filter(Bitmap bitmap, Bitmap ref, int[] colors) {
```

Comment:
This function is really inefficient. If you remember O(n) notation from AP, this function is O(n^4) which is really slow. At worst it should be O(n^2) to iterate over all the pixels. Look into creating a canvas object from the bitmap with with `new Canvas(Bitmap)`. The resulting canvas object will allow you to draw circles much more efficiently. See if you can do this, then mask the resulting bitmap with the ref image. This can probably be done with the techniques described here: https://medium.com/@Tarek360/crop-a-shape-from-an-android-bitmap-9690b7432774. Maybe you can even apply these changes to the other design filters you have created? In a separate pull request of course :)